src/sentry/runner/commands/run.py

```diff
@@ -371,6 +371,7 @@ def post_process_forwarder(**options):
     from sentry.eventstream.base import ForwarderNotRequired
 
     try:
+        # TODO(markus): convert to use run_processor_with_signals -- can't yet because there's a custom shutdown handler
```

Comment:
Would it make sense to support passing a custom function to the `run_processor_with_signals` function as well? Though we don't seem to do it much today, I think shutting down any custom resources the consumer was managing - producers, threadpools, etc - in this handler is something that we should be doing.

Reply:
the custom handler is here: https://github.com/getsentry/sentry/blob/a30619f85fb701d852c2c3d4127302e3114487d1/src/sentry/eventstream/kafka/backend.py#L346-L349

my understanding is that if this is converted to the producer strategy, it won't be necessary anymore

a general hook may make sense but right now i'm just trying to beat all consumers into a similar shape so i can wrap my head around them, the API is not really what I want it to be

Reply:
Arroyo's `produce` strategy takes the producer as an argument, but still leaves it up to the consumer to create and close the resource.